### PR TITLE
[RDBMS] Modified server create with private network, fixed restore time bug

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**
- refactoring private network provision (mysql/postgres flexible-server create --vnet --subnet)
- Handle potential vnet,subnet address prefixes conflicts for create command
- deleted restore time timezone formatting
- improve tests structure for Flexible Server daily end to end test

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by DaeunYim** [comment]
CI test failure due to pytz incompatibility with python 36.. 
Need azp run to initiate the CI test again. Updated the pytz version but might fail as well. 

Will continue fixing this issue tmrw so please go ahead and review this if you happen to get a chance to do..!

File: src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py [hunk 1]
```diff
@@ -297,11 +297,45 @@ def get_current_time():
...
+
+    parent_path = ''
+    if parent_name and parent_type:
+        parent_path = '{}/{}'.format(parent_type, parent_name)
+
+    api_version = _resolve_api_version(resource_client, provider_namespace, resource_type, parent_path)
+
+    try:
+        resource_client.resources.get(resource_group, provider_namespace, parent_path, resource_type, value, api_version)
+    except:  # pylint: disable=bare-except
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py:321]: Could you only catch specific error? If the resource doesn't exist, you will get Azure  related error and get status code 404, right? Could you add such check to return False?

File: src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py [hunk 2]
```diff
@@ -297,11 +297,45 @@ def get_current_time():
...
+def _resolve_api_version(client, provider_namespace, resource_type, parent_path):
+    provider = client.providers.get(provider_namespace)
+
+    # If available, we will use parent resource's api-version
+    resource_type_str = (parent_path.split('/')[0] if parent_path else resource_type)
+
+    rt = [t for t in provider.resource_types  # pylint: disable=no-member
+          if t.resource_type.lower() == resource_type_str.lower()]
+    if not rt:
+        raise CLIError('Resource type {} not found.'.format(resource_type_str))
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py:335]: use error type in azclierror.py https://github.com/Azure/azure-cli/blob/3bff786b9e337aa35bb4cc64a09e5600a35663f8/src/azure-cli-core/azure/cli/core/azclierror.py#L136

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 3]
```diff
@@ -70,16 +58,10 @@ def flexible_server_create(cmd, client, resource_group_name=None, server_name=No
...
-                                 location, DELEGATION_SERVICE_NAME, vnet_address_prefix, subnet_address_prefix)
-        delegated_subnet_arguments = mysql_flexibleservers.models.DelegatedSubnetArguments(
-            subnet_arm_resource_id=subnet_id)
-    elif public_access is None and subnet_arm_resource_id is None and vnet_resource_id is None:
-        subnet_id = create_vnet(cmd, server_name, location, resource_group_name,
-                                DELEGATION_SERVICE_NAME)
-        delegated_subnet_arguments = mysql_flexibleservers.models.DelegatedSubnetArguments(
-            subnet_arm_resource_id=subnet_id)
+    if public_access is None:
+        subnet_id = prepare_private_network(cmd, server_name, vnet_resource_id, subnet_arm_resource_id, resource_group_name,
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:62]: avoid positional arguments

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 4]
```diff
@@ -117,14 +117,13 @@ def _test_flexible_server_create(self, database_engine, resource_group, server):
     def _test_flexible_server_create_non_default_tiers(self, database_engine, resource_group):
 
         if database_engine == 'postgres':
-            self.cmd('postgres flexible-server create -g {} -l {} -n {} --tier Burstable --sku-name Standard_B1ms --public-access none --zone 1'
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:120]: why removing test with --zone?

File: src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py:321]: Yes, added it this! Thanks :)

File: src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py (hunk 2 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_flexible_server_util.py:335]: Modified this.. thanks

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py (hunk 3 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py]: Fixed this. Thanks

**Comment by DaeunYim** [comment]
Hi! Wanted to make sure this change will be checked in for current build :)

**Review by Juliehzl — APPROVED** [review]
Approved as there is a bug to fix

**Comment by DaeunYim** [comment]
> Approved as there is a bug to fix

Thanks! Let me work more in timely manner from next time

**Pull request merged**